allennlp/common/testing/__init__.py

```diff
@@ -45,3 +50,42 @@ def cpu_or_gpu(test_method):
     Decorator to indicate that a test should run on both CPU and GPU
     """
     return pytest.mark.gpu(test_method)
+
+
+# Helpers for testing distributed metrics
+
+
+def assert_metrics_values(
+    metrics: Dict[str, Any], desired_values: Dict[str, Any], exact: Optional[bool] = True
+):
+    for key in metrics:
+        if exact:
+            assert metrics[key] == desired_values[key], "{} != {}".format(
+                metrics[key], desired_values[key]
+            )
+        else:
+            assert_allclose(metrics[key], desired_values[key])
+
+
+def global_distributed_metric(
+    global_rank: int,
+    world_size: int,
+    gpu_id: Union[int, torch.device],
+    metric: Metric,
+    metric_kwargs: Dict[str, Any],
+    desired_values: Dict[str, Any],
```

Comment:
Not really. `metric_kwargs` can be any tensors, or float (in case of perplexity), or even lists of nltk trees. 
 Or do you mean `metric_kwargs` and `desired_values` should be a combined Dict argument to `global_distributed_metric`?

Reply:
Ah, I see what you're saying. As a matter of fact, `metric_kwargs` is supposed to be `Dict[str, List[Any]]`. I'll update the type.